diff --git a/crates/tauri-bundler/src/bundle/linux/appimage/sharun.rs b/crates/tauri-bundler/src/bundle/linux/appimage/sharun.rs
--- a/crates/tauri-bundler/src/bundle/linux/appimage/sharun.rs
+++ b/crates/tauri-bundler/src/bundle/linux/appimage/sharun.rs
@@ -11,10 +11,8 @@ use anyhow::Context;
 
 use crate::{
   bundle::{linux::freedesktop, settings::Arch},
-  utils::{fs_utils, http_utils::download, CommandExt},
+  utils::{fs_utils, CommandExt},
   Settings,
 };
 
-use super::write_and_make_executable;
-
 // TODO: Test if bundling xdg-mime makes sense (eg does it even work if it's not on the host system?)
@@ -60,9 +58,11 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
 
   // TODO: mirror
   let quick_sharun = tools_path.join("quick-sharun.sh");
-  let data = download(
-      "https://raw.githubusercontent.com/pkgforge-dev/Anylinux-AppImages/refs/heads/main/useful-tools/quick-sharun.sh",
-    )?;
-  write_and_make_executable(&quick_sharun, data)?;
+  if !quick_sharun.is_file() {
+    return Err(crate::Error::GenericError(format!(
+      "quick-sharun.sh was not preloaded at {}",
+      quick_sharun.display()
+    )));
+  }
 
   let main_binary = settings.main_binary()?;
